### PR TITLE
fix(account): verify MFA code for every OTP based factors and not only the PENDING_ACTIVATION ones

### PR DESCRIPTION
**Pull request by tcompiegne**
fixes AM-153

### CODE REVIEW

**Review by leleueri — APPROVED** [review]
Two minor changes but otherwise LGTM

File: gravitee-am-common/src/main/java/io/gravitee/am/common/exception/authentication/AuthenticationException.java [hunk 1]
```diff
@@ -52,6 +52,10 @@ public AuthenticationException(String msg, Throwable throwable, Map<String, Stri
 
     public abstract String getErrorCode();
 
+    public int getHttpStatusCode() {
+        return 401;
```
Comment (leleueri) [inline gravitee-am-common/src/main/java/io/gravitee/am/common/exception/authentication/AuthenticationException.java:56]: I would prefer a `HttpStatusCode.UNAUTHORIZED_401`

File: gravitee-am-common/src/main/java/io/gravitee/am/common/exception/mfa/InvalidCodeException.java [hunk 2]
```diff
@@ -31,4 +31,9 @@ public InvalidCodeException(String msg) {
     public String getErrorCode() {
         return "invalid_mfa";
     }
+
+    @Override
+    public int getHttpStatusCode() {
+        return 403;
```
Comment (leleueri) [inline gravitee-am-common/src/main/java/io/gravitee/am/common/exception/mfa/InvalidCodeException.java:37]: I would prefer a `HttpStatusCode.FORBIDDEN_403`

**Review by leleueri — COMMENTED** [review]
Two minor comments but LGTM

File: gravitee-am-common/src/main/java/io/gravitee/am/common/exception/authentication/AuthenticationException.java (hunk 1 above)
Comment (tcompiegne) [reply gravitee-am-common/src/main/java/io/gravitee/am/common/exception/authentication/AuthenticationException.java]: FP

File: gravitee-am-common/src/main/java/io/gravitee/am/common/exception/mfa/InvalidCodeException.java (hunk 2 above)
Comment (tcompiegne) [reply gravitee-am-common/src/main/java/io/gravitee/am/common/exception/mfa/InvalidCodeException.java]: FP

**Pull request merged**